include/chemfiles/FormatFactory.hpp

```diff
@@ -51,16 +61,21 @@ class CHFL_EXPORT FormatFactory final {
     template<class Format>
     void add_format() {
         auto info = format_information<Format>();
-        register_format(info, [](const std::string& path, File::Mode mode, File::Compression compression) {
-            return std::unique_ptr<Format>(new Format(path, mode, compression));  // NOLINT no make_unique in C++11
-        });
+        register_format(info,
+            [](const std::string& path, File::Mode mode, File::Compression compression) {
+                return std::unique_ptr<Format>(new Format(path, mode, compression));  // NOLINT no make_unique in C++11
+            },
+            [](MemoryBuffer& memory, File::Mode mode, File::Compression compression) {
+                return std::unique_ptr<Format>(new Format(memory, mode, compression));  // NOLINT no make_unique in C++11
```

Comment:
I'm not sure how to do this. A constructor taking a `MemoryBuffer` is required, otherwise, the compiler will complain. Are you suggesting a static if or **SNIFAE**?

Reply:
How about two functions `register_format` and `register_format_with_memory`?

Reply:
I still think SNIFAE is better. There could simply be a public tag in the class like `typedef supports_memory_io = int`. Then, there would be two versions of `register_fromat`, one which requires this tag, and another which does not and template substitution will take care of which one to use. The reason I do not like `register_format_with_memory` is that it requires `FormatFactory` to "know" if a format supports this feature, thereby duplicating the number of times this needs to be declared.

Reply:
In fact, we can simply check if the constructor exists.

[https://gist.github.com/thenickdude/5d9995f409418a4a66ac](https://gist.github.com/thenickdude/5d9995f409418a4a66ac)

Reply:
This should be addressed now!